raiden/ui/cli.py

```diff
@@ -1024,8 +1034,19 @@ def _run_app(self):
                 )
         finally:
             self._shutdown_hook()
-            if api_server:
-                api_server.stop()
+
+            def stop_task(task, timeout=None):
+                if isinstance(task, Runnable):
+                    task.stop(timeout=timeout)
+                else:
+                    task.kill(timeout=timeout)
+                task.get()  # re-raise
+
+            stop_tasks = [
```

Comment:
what about changing from an event to an async result?

then the try above could be just use a `event.get()`, and the exception would be re-raised for us automatically

Reply:
Will check how the signals play with it